e2e/common/traits/pod_test.go

```diff
@@ -58,6 +58,30 @@ func TestPodTrait(t *testing.T) {
 	})
 }
 
+func TestPodTraitWithSupplementalGroups(t *testing.T) {
```

Comment:
This pull req is great, but let's not add new `TestPod...` test function to add a test scenario to the pod trait testing. We prefer to add `t.Run("xxx", ...)` to add it to the existing `TestPodTrait()` to avoid create and destroy namespaces frequently for the e2e testing.
https://github.com/apache/camel-k/issues/3298